### ISSUE TITLE
`ruff check src/` does not respect .gitignore for ignored files

### ISSUE THREAD

**Issue by klim-neumann** [astral-sh/ruff#17392]
### Summary

When running `ruff check src/foo`, Ruff correctly respects .gitignore and skips the `_version.py` file, which is generated by [hatch-vcs](https://github.com/ofek/hatch-vcs) and listed in `.gitignore`.

However, when running `ruff check src/`, Ruff stops respecting `.gitignore` and reports lint issues for the same `_version.py` file, even though it is still ignored in Git.

This behavior seems inconsistent and may lead to false-positive reports on ignored/generated files.

**Reproduction Steps:**

1. Create a package `foo` inside `src/` with a file `src/foo/_version.py`.
2. Add `src/foo/_version.py` to `.gitignore`.
3. Run:
`
ruff check src/foo --respect-gitignore  # ✅ _version.py is ignored
ruff check src/ --respect-gitignore     # ❌ _version.py is linted
`

**_version.py**
`
# file generated by setuptools-scm
# don't change, don't track in version control

__all__ = ["__version__", "__version_tuple__", "version", "version_tuple"]

TYPE_CHECKING = False
if TYPE_CHECKING:
    from typing import Tuple
    from typing import Union

    VERSION_TUPLE = Tuple[Union[int, str], ...]
else:
    VERSION_TUPLE = object

version: str
__version__: str
__version_tuple__: VERSION_TUPLE
version_tuple: VERSION_TUPLE

__version__ = version = '0.0.post1.dev8'
__version_tuple__ = version_tuple = (0, 0, 'dev8')
`

**pyproject.toml:**
`
[tool.ruff]
respect-gitignore = true

[tool.ruff.lint]
extend-select = ["I", "INP", "RUF022"]
`

**.gitignore:**
`
src/foo/_version.py
`

### Version

ruff 0.11.5 (7186d5e9a 2025-04-10)

**Comment by ntBre** [comment]
Thank you for the nice writeup! I think this is a duplicate of, or at least closely related to, #13692. However, the point about checking `src/foo` might be a new piece of evidence. The comments I saw in the other issue tested commands like `ruff check .` and `ruff check src` only, so it may work properly when selecting a full path to a directory _or_ no directory/running at the project root, making the problem with partial paths.

**Comment by ntBre** [comment]
I think this was a duplicate of #13692 after all and has now also been resolved! I tested out the example and reproduced the issue on 0.14.2:

```
$ tree
.
├── pyproject.toml
└── src
    └── foo
        └── _version.py

3 directories, 2 files
$ uvx ruff@0.14.2 check src/foo --respect-gitignore
warning: No Python files found under the given path(s)
All checks passed!
$ uvx ruff@0.14.2 check src/ --respect-gitignore
INP001 File `src/foo/_version.py` is part of an implicit namespace package. Add an `__init__.py`.
# ...

Found 2 errors.
[*] 1 fixable with the `--fix` option.
```

and confirmed that it was fixed in 0.14.3 after upgrading our `ignore` dependency:

```
$ uvx ruff@0.14.3 check src/foo --respect-gitignore
warning: No Python files found under the given path(s)
All checks passed!
$ uvx ruff@0.14.3 check src/ --respect-gitignore
warning: No Python files found under the given path(s)
All checks passed!
```
